### PR TITLE
GSYE-526: Make sure that the first week of the DB profile is used whe…

### PR DESCRIPTION
**Pull request by hannesdiedrich**
…n the simulation time doe snot overlap with the profile time

## Reason for the proposed changes

Please describe what we want to achieve and why.

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=bug/GSYE-526
**GSY_FRAMEWORK_BRANCH**=master

### CODE REVIEW

File: src/gsy_e/gsy_e_core/user_profile_handler.py [hunk 1]
```diff
@@ -130,13 +130,17 @@ def get_first_week_from_profile(self, profile_uuid, current_timestamp) -> dict:
...
-                month=current_timestamp.month,
-                day=current_timestamp.day
-            ): datapoint.value for datapoint in datapoints
-        }
+        datapoint_dict = {}
+        diff_current_to_db_time = None
+        for datapoint in datapoints:
+            db_time = self._strip_timezone_and_create_pendulum_instance_from_datetime(
+                datapoint.time)
+            if diff_current_to_db_time is None:
```
Comment (mrsaemir) [inline src/gsy_e/gsy_e_core/user_profile_handler.py:138]: I'd suggest to move this before the `for` loop as it's calculated only once with the first element.

Comment (hannesdiedrich) [reply src/gsy_e/gsy_e_core/user_profile_handler.py:138]: I Ieft a verbose comment one line below in order to explain why it is here. Let me know if it is not clear.

Comment (mrsaemir) [reply src/gsy_e/gsy_e_core/user_profile_handler.py:138]: Yes. I do get it. However, you may be a able to evaluate the query before the for loop.
```python
datapoints = list(datapoints)
```
I guess this way you are not refetching the whole set from the DB again.

Comment (hannesdiedrich) [reply src/gsy_e/gsy_e_core/user_profile_handler.py:138]: Good idea, will change this!

Comment (hannesdiedrich) [reply src/gsy_e/gsy_e_core/user_profile_handler.py]: Done [890adb1](https://github.com/gridsingularity/gsy-e/pull/1603/commits/890adb1115f6135eba8843a91a4667ecb38c187e)

**Review by mrsaemir — APPROVED** [review]
LGTM. Thanks for dealing with this.

**Pull request merged**